### PR TITLE
Enable multi-select

### PR DESCRIPTION
**Pull request by hexium310**

### CODE REVIEW

**Comment by chitoku-k** [comment]
I think many of our completions should actually enable this feature without making them optional.

**Comment by chitoku-k** [comment]
The subcommands listed below accept multiple parameters to be passed.
- add
- branch
- cherry-pick
- commit
  - `--fixup` does not accept multiple parameters
- merge

File: git.zsh [hunk 1]
```diff
@@ -69,7 +69,7 @@ _fzf_complete_git-commit-messages_post() {
 }
 
 _fzf_complete_git-unstaged-files() {
-    _fzf_complete '--ansi' "$@" < <(git status --porcelain=v1 2> /dev/null | awk \
```
Comment (chitoku-k) [inline git.zsh]: Tips: using `shift` makes this much more readable

```zsh
_fzf_complete_git-unstaged-files() {
    local options="$1"
    shift
    _fzf_complete "--ansi $options" "$@" # ...
}
```
```

**Comment by hexium310** [comment]
> The subcommands listed below accept multiple parameters to be passed.
> 
> * add
> * branch
> * cherry-pick
> * commit
>   
>   * `--fixup` does not accept multiple parameters
> * merge

Which implement multi-select to these subcommands themselves or to options that are passed multiple parameters?

**Comment by chitoku-k** [comment]
Too hard to consider the order of parameters, just implementing for sucommands that supports multiple parameters is sufficient.

**Comment by hexium310** [comment]
The multiple parameters that `git commit` is passed are the files in the work tree.
However, the completion of `git commit` shows commit message, so the completion role is different.

**Comment by chitoku-k** [comment]
Absolutely correct. The current completion for `git commit` is totally wrong as it does't suggest as you described. How about making the commit messages appear only after `-m` or `--message`?

**Comment by hexium310** [comment]
Below two commands are available, so
> Too hard to consider the order of parameters, just implementing for subcommands that support multiple parameters is sufficient.
- `git commit <files> -m <message>`
- `git commit -m <message> -- <files>`

**Comment by chitoku-k** [comment]
Can you also enable all the other completion functions to accept an fzf option as its first parameter?

**Comment by chitoku-k** [comment]
Very nice!

**Pull request merged**